src/tests/Microsoft.PowerFx.Interpreter.Tests/UserDefinedTests.cs

```diff
@@ -24,5 +27,52 @@ public void NamedFormulaEntryTest(string script, string expression, double expec
             var result = (NumberValue)check.GetEvaluator().Eval();
             Assert.Equal(expected, result.Value);
         }
+
+        [Theory]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) }; num = 3;")]
+        public void ValidUDFBodyTest(string script)
+        {
+            var options = new ParserOptions()
+            {
+                AllowsSideEffects = true,
+                Culture = CultureInfo.InvariantCulture
+            };
+            UserDefinitions.ProcessUserDefinitions(script, options, out var result);
+
+            Assert.False(result.HasErrors);
+        }
```

Comment:
More interesting tests:

- F1(x:Number): Number = { Set(a, x); x+1; };
  - behavior UDF with a return value
- F1(x:Number): Boolean = { Set(a, x); Today(); };
  - That should be an error - we cannot coerce a date to a Boolean value